lrpar/examples/calc_wasm/src/lib.rs

```diff
@@ -0,0 +1,117 @@
+#[cfg(all(
+    target_arch = "wasm32",
+    target_os = "unknown",
+    target_vendor = "unknown"
+))]
+use wasm_bindgen::prelude::*;
+
+use lrlex::lrlex_mod;
+use lrpar::lrpar_mod;
+
+// Using `lrlex_mod!` brings the lexer for `calc.l` into scope. By default the module name will be
+// `calc_l` (i.e. the file name, minus any extensions, with a suffix of `_l`).
+lrlex_mod!("calc.l");
+// Using `lrpar_mod!` brings the parser for `calc.y` into scope. By default the module name will be
+// `calc_y` (i.e. the file name, minus any extensions, with a suffix of `_y`).
+lrpar_mod!("calc.y");
+
+#[cfg_attr(
+    all(
+        target_arch = "wasm32",
+        target_os = "unknown",
+        target_vendor = "unknown"
+    ),
+    wasm_bindgen
+)]
+pub fn calculate(l: &str) -> Result<u64, String> {
+    // Get the `LexerDef` for the `calc` language.
+    let lexerdef = calc_l::lexerdef();
+    if l.trim().is_empty() {
+        return Err("input is empty".to_string().into());
+    }
+    // Now we create a lexer with the `lexer` method with which we can lex an input.
+    let lexer = lexerdef.lexer(&l);
+    // Pass the lexer to the parser and lex and parse the input.
+    let (res, errs) = calc_y::parse(&lexer);
+    if !errs.is_empty() {
+        let mut ret = String::new();
+        for e in errs {
+            use lrpar::LexParseError;
+            match e {
+                LexParseError::ParseError(e) => {
+                    let repairs_flag = !e.repairs().is_empty();
+                    ret.push_str(&format!(
+                        "Error: {}\n Repairs: {}",
+                        e.to_string(),
+                        repairs_flag
+                    ));
+                }
+                e => ret.push_str(&format!("{}\n", e.to_string())),
+            };
+        }
+        if let Some(Err(e)) = res {
+            ret.push_str(&format!("{}\n", e.to_string()));
+        }
+        return Err(ret);
+    }
+    match res {
+        Some(Ok(r)) => Ok(r),
+        Some(Err(e)) => Err(e.to_string()),
+        None => Err("Unable to parse".to_string().into()),
+    }
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
I don't think i've ever actually used a `test` folder but I don't see any way to specify a crate type for them.
Because of how `wasm32-unknown-unknown` works, this seems to need to be a `cdylib` crate.

the `#[cfg(test)]` module is used for `wasm32-wasip2` and native targets,
while the other section is used only for `wasm32-unknown-unknown`.

Reply:
Should be fixed by moving it into cttests in [c5a4efd](https://github.com/softdevteam/grmtools/pull/527/commits/c5a4efdebab5ee4148203551bdfb1baefec64aab)